### ISSUE TITLE
Adding the same unit a second time is not prohibited on db level

### ISSUE THREAD

**Issue by SuperMasterPhoenix** [TandoorRecipes/recipes#1085]
### Version
Please provide your current version (can be found on the system page since v0.8.4)
Version: 1.0.1

### Bug description
Somehow I managed to add the same unit twice. I'm not quite sure how I did it, maybe I was too fast when adding an ingredient and accidentally hit `create` before the list of units was loaded. See this database screenshot:
![grafik](https://user-images.githubusercontent.com/20322801/143057028-2b229b53-b04c-4f9d-befd-faa5d8483bcb.png)

This led to not being able to edit or add recipes where the ingredients use this doublicated unit. Doing so resulted in a http 500 error:
> MultipleObjectsReturned at /api/recipe/129/
get() returned more than one Unit -- it returned 2!

I could solve the issue by merging the two units using the ui.

A possible solution could be to add a unique key database constraint to the name column of the table `cookbook_unit` to prevent this issue.

**Comment by vabene1111** [comment]
``Thanks for the feedback. Is it possible that you might have used the admin to create the unit ? 

Because there is technically a constraint on the unit model (at least in the ORM but that is used everywhere) that should prevent this from happening. We had some issues with a leading or trailing space on the unit quite a while back but it has not happend for a long time.

```python
class Unit(ExportModelOperationsMixin('unit'), models.Model, PermissionModelMixin):
    name = models.CharField(max_length=128, validators=[MinLengthValidator(1)])
    description = models.TextField(blank=True, null=True)

    space = models.ForeignKey(Space, on_delete=models.CASCADE)
    objects = ScopedManager(space='space')

    def __str__(self):
        return self.name

    class Meta:
        constraints = [
            models.UniqueConstraint(fields=['space', 'name'], name='u_unique_name_per_space')
        ]
```

Can you roughly describe what you did or do you have any idea what you could have done, maybe there is some function somewhere that ignores the constraint for some reason.

**Comment by SuperMasterPhoenix** [comment]
>  Is it possible that you might have used the admin to create the unit ?

No, this was definitively with my non-admin user because I was adding a recipe and suddenly I couldn't change it anymore. And I didn't use the admin user for quite a while now.

~~I already tried to reproduce the issue but without success. As I said I think I was adding ingredients to a recipe, using tab between the amount and unit field and (maybe) typing a 'g' in the unit field followed by pressing enter (-> create the unit) before all suggestions were loaded. But I don't know whether it really happened that way.~~

I just checked the database again and as you mentioned there is already the `u_unique_name_per_space` constraint. Then I tried to manually insert the already existing data `insert into public.cookbook_unit values (87, 'g', null, 1)` and it worked despite the constraint:
![grafik](https://user-images.githubusercontent.com/20322801/143099019-dcf46385-7b8b-4d0b-beb1-38b3fec43029.png)

Interestingly a second insert `insert into public.cookbook_unit values (88, 'g', null, 1)` now fails as expected:
```
ERROR:  duplicate key value violates unique constraint "u_unique_name_per_space"
DETAIL:  Key (space_id, name)=(1, g) already exists.
SQL state: 23505
```

I manually deleted and recreated the constraint in the database and now the insert already fails on the first try - as expected. Maybe this is caused by the postgres upgrade from 13 to 14 I did yesterday. Sorry for bringing this up to you, I think I'll have to do some sort of index-cleanup on my databases.
